### PR TITLE
[#4370]feat(iceberg): support view interface for Iceberg REST server

### PR DESCRIPTION
**Pull request by theoryxu**
### What changes were proposed in this pull request?

support view interface for Iceberg REST server

### Why are the changes needed?

Fix: #4370

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

1. add UT
2. manual test

### CODE REVIEW

**Comment by theoryxu** [comment]
@FANNG1  please help to review, Thanks.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergViewOperations.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    ResourceConfig resourceConfig =
+        IcebergRestTestUtil.getIcebergResourceConfig(IcebergViewOperations.class);
+    // create namespace before each view test
+    resourceConfig.register(IcebergNamespaceOperations.class);
+    resourceConfig.register(IcebergViewRenameOperations.class);
+
+    return resourceConfig;
+  }
+
+  private Response doCreateView(String name) {
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergViewOperations.java:70]: Could you place the private method to the last of class?

**Comment by FANNG1** [comment]
@theoryxu , thanks for the PR,  after supporting views, could Spark create the view in the Iceberg REST catalog?

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergViewOperations.java (hunk 1 above)
Comment (theoryxu) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergViewOperations.java:70]: ok

**Comment by theoryxu** [comment]
> @theoryxu , thanks for the PR, after supporting views, could Spark create the view in the Iceberg REST catalog? If yes, could you add corresponding integration test?

Sure, I tested it in an internal environment and will add an integration test soon.

**Comment by theoryxu** [comment]
@FANNG1 Integration tests have been added. Please review. Thanks!

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java [hunk 2]
```diff
@@ -557,4 +558,142 @@ void testRegisterTable() {
     result = convertToStringMap(sql("SELECT * FROM iceberg_rest_table_test.register_foo2"));
     Assertions.assertEquals(ImmutableMap.of("1", "a", "2", "b"), result);
   }
+
+  @Test
+  @EnabledIf("catalogTypeNotHive")
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java:563]: could you use `isSupportsViewCatalog` ?

**Comment by FANNG1** [comment]
LGTM except one comment, @theoryxu could you fix the ci error?

**Review by FANNG1 — APPROVED** [review]

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTJdbcCatalogIT.java [hunk 3]
```diff
@@ -68,6 +68,8 @@ public Map<String, String> getCatalogConfig() {
     configMap.put(
         IcebergConfig.ICEBERG_CONFIG_PREFIX + IcebergConfig.JDBC_INIT_TABLES.getKey(), "true");
 
+    configMap.put(IcebergConfig.ICEBERG_CONFIG_PREFIX + "jdbc.schema-version", "V1");
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTJdbcCatalogIT.java:71]: should we enable this configuration by default, does it have some bad effect?

Comment (theoryxu) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTJdbcCatalogIT.java:71]: This configuration will alter the metadata schema of JdbcCatalog. Perhaps it's preferable to allow users to set this explicitly? They should be aware of the exact changes made.

We could introduce a `jdbc.schema-version` property in IcebergConfig and document its usage ？

Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTJdbcCatalogIT.java:71]: +1 for allowing user setting this explicitly.  seems no need to add `jdbc.schema-version` in IcebergConfig, do you like to add related document in this PR or a separate PR?

File: docs/iceberg-rest-service.md [hunk 4]
```diff
@@ -190,16 +190,17 @@ The Gravitino Iceberg REST catalog service uses the memory catalog backend by de
...
+| `gravitino.iceberg-rest.jdbc.password`                         | The password of the JDBC connection.                                                                                                         | (none)                  | Yes      | 0.2.0         |
+| `gravitino.iceberg-rest.jdbc-initialize`                       | Whether to initialize the meta tables when creating the JDBC catalog.                                                                        | `true`                  | No       | 0.2.0         |
+| `gravitino.iceberg-rest.jdbc-driver`                           | `com.mysql.jdbc.Driver` or `com.mysql.cj.jdbc.Driver` for MySQL, `org.postgresql.Driver` for PostgreSQL.                                     | (none)                  | Yes      | 0.3.0         |
+| `gravitino.iceberg-rest.jdbc.schema-version`                   | JDBC catalog is initialized without view support. To auto-migrate the database's schema and enable view support, set jdbc.schema-version=V1  | (none)                  | NO       | 0.7.0         |
```
Comment (FANNG1) [inline docs/iceberg-rest-service.md:203]: Could you add a separate section for view support?

File: docs/iceberg-rest-service.md [hunk 5]
```diff
@@ -215,6 +214,15 @@ You must download the corresponding JDBC driver to the `iceberg-rest-server/libs
 
 If you want to use a custom Iceberg Catalog as `catalog-backend`, you can add a corresponding jar file to the classpath and load a custom Iceberg Catalog implementation by specifying the `catalog-backend-impl` property.
 
+#### View support
+
+The Gravitino Iceberg REST server supports view interfaces through its JDBC backend. You can enable this by setting the `jdbc.schema-version` property.
+
+| Configuration item                              | Description                                                                                                                                  | Default value           | Required | Since Version |
```
Comment (FANNG1) [inline docs/iceberg-rest-service.md:221]: Could you change the description like that? 
```
The schema version of JDBC catalog backend, setting to `V1` if supporting view operations. 
```

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@theoryxu , merged to main, thanks for your contribution.

**Pull request merged**